packages/salesforce-adapter/test/adapter.crud.test.ts

```diff
@@ -1417,298 +1486,344 @@ describe('SalesforceAdapter CRUD', () => {
   describe('Update operation', () => {
     let result: DeployResult
     describe('for an instance element', () => {
-      let beforeInstance: InstanceElement
-      let afterInstance: InstanceElement
-
-      beforeEach(() => {
-        beforeInstance = createInstanceElement(mockDefaultValues.Profile, mockTypes.Profile)
-        afterInstance = createInstanceElement(
-          {
-            ...mockDefaultValues.Profile,
-            description: 'Updated profile description',
-          },
-          mockTypes.Profile,
-        )
-      })
-
-      describe('when the request succeeds', () => {
-        const DEPLOYMENT_ID = 'testDeploymentId'
+      describe('for non profile instances', () => {
         beforeEach(async () => {
+          const DEPLOYMENT_ID = 'testDeploymentId'
+          const beforeInstanceNoProfile = createInstanceElement(mockDefaultValues.ApexPage, mockTypes.ApexPage)
+          const afterInstanceNoProfile = createInstanceElement(
+            {
+              ...mockDefaultValues.ApexPage,
+              description: 'Updated ApexPage description',
+            },
+            mockTypes.ApexPage,
+          )
           connection.metadata.deploy.mockReturnValueOnce(
             mockDeployResult({
               id: DEPLOYMENT_ID,
               success: true,
               componentSuccess: [
                 {
-                  fullName: mockDefaultValues.Profile.fullName,
-                  componentType: constants.PROFILE_METADATA_TYPE,
+                  fullName: mockDefaultValues.ApexPage.fullName,
+                  componentType: constants.APEX_PAGE_METADATA_TYPE,
+                  id: 'new_id',
                 },
               ],
               retrieveResult: await mockRetrieveResult({}),
             }),
           )
           result = await adapter.deploy({
             changeGroup: {
-              groupID: afterInstance.elemID.getFullName(),
+              groupID: afterInstanceNoProfile.elemID.getFullName(),
               changes: [
                 {
                   action: 'modify',
-                  data: { before: beforeInstance, after: afterInstance },
+                  data: { before: beforeInstanceNoProfile, after: afterInstanceNoProfile },
                 },
               ],
             },
             progressReporter: nullProgressReporter,
           })
         })
-        it('should return an InstanceElement', () => {
-          expect(result.appliedChanges).toHaveLength(1)
-          expect(getChangeData(result.appliedChanges[0])).toBeInstanceOf(InstanceElement)
+        it('should update internal id for non profile instances', async () => {
+          const a = _.get(getChangeData(result.appliedChanges[0]), 'value')[constants.INTERNAL_ID_FIELD]
+          expect(a).not.toBe('ApexPageId')
+          expect(a).toBe('new_id')
```

Comment:
```suggestion
          const deployedInstance = getChangeData(result.appliedChanges[0]) as InstanceElement
          expect(deployedInstance).toBeInstanceOf(InstanceElement)
          expect(deloyedInstance.value[constants.INTERNAL_ID_FIELD]).toEqual('new_id')
```
What is `a` 🤔 ?
Consider the following suggestion